content/specs/auth.md

```diff
@@ -0,0 +1,390 @@
+
+The Authorization Server uses the `request_uri` to look up the earlier Authorization Request parameters, authenticates the user (which might include sign-in or account selection), and prompts the user with the Authorization Interface. The user might refine any granular requested scopes, then approves or rejects the request. The Authorization Server redirects the user back to the `redirect_uri`, which might be a web callback URL (for web clients), or a native app URI.
+
+The client uses URL query parameters (`state` and `iss`) to look up and verify session information and stores the authorization `code`. Using the `code`, the client then makes an initial token request to the Authorization Server’s token endpoint. The client completes the PKCE flow by including the earlier value in the `code_verifier` field. Confidential clients need to include a client assertion JWT in the token request; see the "Confidential Client" section. The Authorization Server validates the request and returns a set of tokens, as well as a `sub` field indicating the account identifier (DID) for this session.
+
+At this point it is critical (mandatory) for all clients to verify that the account identified by the `sub` field is consistent with the Authorization Server `issuer`, either by validating against the originally-supplied account DID, or by resolving the accounts DID to confirm the PDS is consistent with the Authorization Server. See “Identity Authentication” section.
+
+TODO: how does the client know what scopes it got approved for? should we require that scope info gets passed back in the token response?
+
+Authentication-only clients can end the flow here.
+
+Using the access token, clients are now able to make authorized requests to the PDS (”Resource Server”). They must use DPoP for all such requests, along with the access token. Tokens (both access and refresh) will need to be periodically “refreshed” by subsequent request to the Authorization Server token endpoint. These also require DPoP. See “Tokens and Session Lifetime” section for details.
+
+TODO: users should be able to revoke session from the PDS web interface
+
+### OAuth Security Considerations
+
+There are a number of situations where HTTP URLs provided by external parties are fetched by both clients and providers (servers). Care must be taken to prevent harmful fetches due to maliciously crafted URLs, including hostnames which resolve to private or internal IP addresses. The general term for this class of security issue is SSRF. There is also a class of denial-of-service attacks involving HTTP requests to malicious servers, such as huge response bodies, TCP-level slow-loris attacks, etc. We strongly recommend using "hardened" HTTP client implementations/configurations to mitigate these attacks.
```

Comment:
```suggestion
There are a number of situations where HTTP URLs provided by external parties are fetched by both clients and providers (servers). Care must be taken to prevent harmful fetches due to maliciously crafted URLs, including hostnames which resolve to private or internal IP addresses. The general term for this class of security issue is Server-Side Request Forgery (SSRF). There is also a class of denial-of-service attacks involving HTTP requests to malicious servers, such as huge response bodies, TCP-level slow-loris attacks, etc. We strongly recommend using "hardened" HTTP client implementations/configurations to mitigate these attacks.
```
Also: might be helpful to spell out what SSRF means here